src/Workspaces/Core/Portable/CodeActions/CodeAction.cs

```diff
@@ -236,8 +237,14 @@ protected virtual async Task<IEnumerable<CodeActionOperation>> ComputeOperations
             return new CodeActionOperation[] { new ApplyChangesOperation(changedSolution) };
         }
 
-        internal virtual async Task<ImmutableArray<CodeActionOperation>> ComputeOperationsAsync(
-            IProgressTracker progressTracker, CancellationToken cancellationToken)
+        /// <summary>
+        /// Override this method if you want to implement a <see cref="CodeAction"/> subclass that includes custom <see
+        /// cref="CodeActionOperation"/>'s.  Prefer overriding this method over <see
+        /// cref="ComputeOperationsAsync(CancellationToken)"/> when computation is long running and progress should be
+        /// shown to the user.
+        /// </summary>
+        private protected virtual async Task<ImmutableArray<CodeActionOperation>> ComputeOperationsAsync(
+            IProgress<CodeAnalysisProgress> progress, CancellationToken cancellationToken)
```

Comment:
this is a method that would become purely protected once we agree that thsi api shape looks good.